projects/core/src/site-context/facade/base-site.service.ts

```diff
@@ -91,4 +122,14 @@ export class BaseSiteService implements SiteContext<BaseSite> {
       getContextParameterDefault(this.config, BASE_SITE_CONTEXT_ID)
     );
   }
+
+  private isCurrentBaseSite(site: BaseSite): boolean {
+    const index = (site.urlPatterns || []).findIndex((jsRegexp: any) => {
+      if (jsRegexp) {
+        return jsRegexp.test(this.currentUrl || '');
+      }
+    });
+
+    return index !== -1;
+  }
```

Comment:
I'm in favour of moving this logic to `SiteContextConfigLoaderService`. It's not needed anytime after we initialize the sitecontext config. It will also help avoiding additional complexity in `get()`.

Any objections?

Reply:
if move this logic to `SiteContextConfigLoaderService`, then we have to call `getAll()` in ``SiteContextConfigLoaderService`. And, we may also have other `configLoaderService`, if they need baseSite data, they also cannot  call `get()`.